pallets/msa/src/lib.rs

```diff
@@ -1655,6 +1643,16 @@ impl<T: Config + Send + Sync> CheckFreeExtrinsicUse<T> {
 			key_count == 1,
 			InvalidTransaction::Custom(ValidityError::InvalidMoreThanOneKeyExists as u8)
 		);
+
+		let has_delegations: bool =
+			DelegatorAndProviderToDelegation::<T>::iter_key_prefix(DelegatorId(msa_id))
```

Comment:
ohh are you sure this does not invoke a read per item, which is probably not being caught by the benchmarks?

Reply:
This was @wilwade's addition.  What do you think @wilwade ?

Reply:
It might not be a bad idea to update this storage to a map count thing that Wil found.

Reply:
Agreed but that is dependent on upgrading Polkadot.  I think it's in 0.9.38.

Reply:
After reviewing `count()` vs `any()` with @enddynayn,` any()` does one additional read while count calls `next()` repeatedly, counting the number of iterations.